api/src/main/java/org/apache/iceberg/transforms/Timestamps.java

```diff
@@ -31,54 +33,159 @@
 import org.apache.iceberg.util.DateTimeUtil;
 import org.apache.iceberg.util.SerializableFunction;
 
-enum Timestamps implements Transform<Long, Integer> {
-  YEAR(ChronoUnit.YEARS, "year"),
-  MONTH(ChronoUnit.MONTHS, "month"),
-  DAY(ChronoUnit.DAYS, "day"),
-  HOUR(ChronoUnit.HOURS, "hour");
+class Timestamps implements Transform<Long, Integer> {
```

Comment:
Looks like this converts Timestamps from an enum into a class. That won't work when using identity comparisons in code like the `SortOrderVisitor` above.